File: ReactJS-Spring-Boot-CRUD-Full-Stack-App/react-frontend/src/services/DepartmentService.js
import axiosClient from '../api/axiosClient';

class DepartmentService {
    getDepartments() {
        return axiosClient.get('/departments');
    }

    getDepartmentById(id) {
        return axiosClient.get(`/departments/${id}`);
    }

    createDepartment(department) {
        return axiosClient.post('/departments', department);
    }

    updateDepartment(id, department) {
        return axiosClient.put(`/departments/${id}`, department);
    }

    deleteDepartment(id) {
        return axiosClient.delete(`/departments/${id}`);
    }
}

export default new DepartmentService();

File: ReactJS-Spring-Boot-CRUD-Full-Stack-App/springboot-backend/src/main/java/net/javaguides/springboot/service/DepartmentService.java
package net.javaguides.springboot.service;

import java.util.List;

import net.javaguides.springboot.dto.DepartmentRequestDto;
import net.javaguides.springboot.dto.DepartmentResponseDto;

public interface DepartmentService {

    List<DepartmentResponseDto> getAllDepartments();

    DepartmentResponseDto getDepartmentById(Long id);

    DepartmentResponseDto createDepartment(DepartmentRequestDto requestDto);

    DepartmentResponseDto updateDepartment(Long id, DepartmentRequestDto requestDto);

    void deleteDepartment(Long id);
}
